crates/project/src/lsp_store.rs

```diff
@@ -11379,7 +11464,7 @@ fn subscribe_to_binary_statuses(
 fn lsp_workspace_diagnostics_refresh(
     server: Arc<LanguageServer>,
     cx: &mut Context<'_, LspStore>,
-) -> Option<(mpsc::Sender<()>, Task<()>)> {
+) -> Option<(mpsc::Sender<()>, mpsc::Sender<()>, Task<()>)> {
```

Comment:
At this point, should we introduce a struct with named fields for all this?
It is not very clear what each channel is doing by looking at this.